src/py/flwr/client/app.py

```diff
@@ -459,6 +461,10 @@ def _on_backoff(retry_state: RetryState) -> None:
 
                     run.fab_id, run.fab_version = fab_id, fab_version
 
+                    # Set node_id for this run
+                    if node_id is not None:
+                        state.set_node_id(node_id)
```

Comment:
The `node_id` is not related to the run. The `node_id` of a SuperNode never changes.

I'd suggest to store the `node_id` in `NodeState` right after we get it from `create_node`. With that, we can remove the local `node_id` variable and read `node_id` from `NodeState` whenever we need it.

Reply:
Fixed.